packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx

```diff
@@ -144,6 +133,26 @@ export const useGridColumnHeaders = (props: UseGridColumnHeadersProps) => {
     [],
   );
 
+  const leftRenderContext = React.useMemo(() => {
+    return renderContext && pinnedColumns.left.length
+      ? {
+          ...renderContext,
+          firstColumnIndex: 0,
+          lastColumnIndex: pinnedColumns.left.length,
+        }
+      : null;
+  }, [renderContext, pinnedColumns.left.length]);
```

Comment:
Ah, ok.
I'll create a separate interface for columns render context then (instead of setting row indices to -1).
This way it's clear that row indices aren't there and you need to use a different selector if you need them.

Reply:
Done